tests/lib/rules/no-side-effects.js

```diff
@@ -302,5 +327,14 @@ eslintTester.run('no-side-effects', rule, {
       output: null,
       errors: [{ message: ERROR_MESSAGE, type: 'CallExpression' }],
     },
+
+    {
+      // checkPlainGetters = true
+      code:
+        'import Component from "@ember/component"; class Foo extends Component { get foo() { this.x = 123; } }',
+      output: null,
+      options: [{ checkPlainGetters: true }],
+      errors: [{ message: ERROR_MESSAGE, type: 'AssignmentExpression' }],
+    },
```

Comment:
Shall we also check for `set(this, 'x', 123)`?

While we are doing the Octane migration, sometimes we have to set non-tracked value because some nested internal Component/EmberObjects are still classic style.

Reply:
That will be covered by this change (there are existing test cases to verify the side effect check include that).

Reply:
Great. Thanks for confirming.